File: javaclient/app/src/main/java/javaclient/app/specs/item/WithProviderItemRequestBuilder.java
package javaclient.client.specs.item;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
import javaclient.client.specs.item.item.ApiJsonItemRequestBuilder;
/**
 * Builds and executes requests for operations under /specs/{provider}
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class WithProviderItemRequestBuilder extends BaseRequestBuilder {
    /**
     * Gets an item from the javaclient.client.specs.item.item collection
     * @param apiJsonId Unique identifier of the item
     * @return a {@link ApiJsonItemRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ApiJsonItemRequestBuilder byApiJsonId(@jakarta.annotation.Nonnull final String apiJsonId) {
        Objects.requireNonNull(apiJsonId);
        final HashMap<String, Object> urlTplParams = new HashMap<String, Object>(this.pathParameters);
        urlTplParams.put("apiJson%2Did", apiJsonId);
        return new ApiJsonItemRequestBuilder(urlTplParams, requestAdapter);
    }
    /**
     * Instantiates a new {@link WithProviderItemRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public WithProviderItemRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/specs/{provider}", pathParameters);
    }
    /**
     * Instantiates a new {@link WithProviderItemRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public WithProviderItemRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/specs/{provider}", rawUrl);
    }
}

File: goclient/specs/with_provider_item_request_builder.go
package specs

import (
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
)

// WithProviderItemRequestBuilder builds and executes requests for operations under \specs\{provider}
type WithProviderItemRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// ByApiJsonId gets an item from the github.com/andrueastman/UntypedNodeSamples/GoClient/.specs.item.item collection
// returns a *ItemApiJsonItemRequestBuilder when successful
func (m *WithProviderItemRequestBuilder) ByApiJsonId(apiJsonId string)(*ItemApiJsonItemRequestBuilder) {
    urlTplParams := make(map[string]string)
    for idx, item := range m.BaseRequestBuilder.PathParameters {
        urlTplParams[idx] = item
    }
    if apiJsonId != "" {
        urlTplParams["apiJson%2Did"] = apiJsonId
    }
    return NewItemApiJsonItemRequestBuilderInternal(urlTplParams, m.BaseRequestBuilder.RequestAdapter)
}
// NewWithProviderItemRequestBuilderInternal instantiates a new WithProviderItemRequestBuilder and sets the default values.
func NewWithProviderItemRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*WithProviderItemRequestBuilder) {
    m := &WithProviderItemRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/specs/{provider}", pathParameters),
    }
    return m
}
// NewWithProviderItemRequestBuilder instantiates a new WithProviderItemRequestBuilder and sets the default values.
func NewWithProviderItemRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*WithProviderItemRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewWithProviderItemRequestBuilderInternal(urlParams, requestAdapter)
}
